packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -257,8 +257,21 @@ class FlutterErrorDetails {
 
 typedef ErrorBuilderCallback<B extends FlutterErrorBuilder> = B Function();
 
+/// Helper class used for collecting different pieces of information
+/// for constructing an instance of [FlutterError]. It provides a number
+/// of methods with names starting with 'add' to makes it
+/// convenient to add different parts to an error report. 
+/// 
+/// See also:
+/// 
+/// * [WidgetErrorBuilder], which adds a few ready-made error elements 
+/// for reporting errors at the widget layer
+/// * [RenderErrorBuilder], which adds a few ready-made error elements 
+/// for reporting errors at the rendering layer
 class FlutterErrorBuilder {
+  // TODO(inmatrix): ask what _buildErrorCallback is about
```

Comment:
it is about when you need to be lazy about building the error details s building does something expensive like parse stack traces. Replaces existing lazy cases.

Reply:
Thanks. I added the following comments:
```
Creates a [FlutterErrorBuilder] with its details computed only when needed.
Use if computing the error details may throw an exception or is expensive.
```